extensions/NexusKitten/moremotion.js

```diff
@@ -0,0 +1,294 @@
+            }
+          },
+          {
+            filter: [Scratch.TargetType.SPRITE],
+            opcode: 'inbetween',
+            blockType: Scratch.BlockType.BOOLEAN,
+            text: 'in-between x1: [X1] y1: [Y1] x2: [X2] y2: [Y2]?',
+            arguments: {
+              X1: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: '-100'
+              },
+              Y1: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: '-100'
+              },
+              X2: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: '100'
+              },
+              Y2: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: '100'
+              }
+            }
+          },
+        ],
+        menus: {
+          WHAT: {
+            acceptreporters: false,
+            items: ['width', 'height']
+          }
+        }
+      };
+    }
+
+    changexy(args, util) {
+      const x = Scratch.Cast.toNumber(args.X);
+      const y = Scratch.Cast.toNumber(args.Y);
+      util.target.setXY(util.target.x + x, util.target.y + y);
+    }
+
+    // LORAX APPROVED
+    pointto(args, util) {
+      const x = Scratch.Cast.toNumber(args.X);
+      const y = Scratch.Cast.toNumber(args.Y);
+      if (util.target.y > y) {
+        util.target.setDirection(((180 / Math.PI) * Math.atan((x - util.target.x) / (y - util.target.y))) + 180);
+      } else {
+        util.target.setDirection(((180 / Math.PI) * Math.atan((x - util.target.x) / (y - util.target.y))));
+      }
+    }
+
+    rotationStyle(args, util) {
+    return util.target.rotationStyle;
+    }
+
+    fence(util) {
+      util.target.setXY(Math.min(Math.max(util.target.x, -Scratch.vm.runtime.stageWidth / 2), Scratch.vm.runtime.stageWidth / 2), Math.min(Math.max(util.target.y, -Scratch.vm.runtime.stageHeight / 2), Scratch.vm.runtime.stageHeight / 2));
+    }
+
+    directionto(args, util) {
+      const x = Scratch.Cast.toNumber(args.X);
+      const y = Scratch.Cast.toNumber(args.Y);
+      if (util.target.y > y) {
+        return ((180 / Math.PI) * Math.atan((x - util.target.x) / (y - util.target.y))) + 180;
+      } else {
+        return ((180 / Math.PI) * Math.atan((x - util.target.x) / (y - util.target.y)));
+      }
+    }
+
+    distanceto(args, util) {
+      const x = Scratch.Cast.toNumber(args.X);
+      const y = Scratch.Cast.toNumber(args.Y);
+      // Shoutout to Pythagoras!
+      return Math.sqrt(((x - util.target.x) ** 2) + ((y - util.target.y) ** 2));
+    }
+
+    steptowards(args, util) {
+      const x = Scratch.Cast.toNumber(args.X);
+      const y = Scratch.Cast.toNumber(args.Y);
+      const steps = Scratch.Cast.toNumber(args.STEPS);
+      const val = steps / (Math.sqrt(((x - util.target.x) ** 2) + ((y - util.target.y) ** 2)));
+      util.target.setXY(((x - util.target.x) * (val)) + util.target.x, ((y - util.target.y) * (val)) + util.target.y);

```

Comment:
If the distance to the point is less than the distance to step, wouldn't it make sense to not step too far?